alicloud/pipeline.yml

```diff
@@ -0,0 +1,306 @@
+          delete_on_failure: true
+          terraform_source: pipelines/alicloud/assets/terraform
+      - do:
+        - <<: *prepare-director
+          params:
+            <<: *prepare-director-params
+            OPTIONAL_OPS_FILE:  |
+              -o pipelines/shared/assets/ops/remove-hm.yml
+              -o bosh-deployment/external-ip-with-registry-not-recommended.yml
+              -o pipelines/shared/assets/ops/remove-provider-cert.yml
+        - do:
+          - <<: *deploy-director
+          - <<: *run-bats
+            params:
+              <<: *run-bats-params
+              STEMCELL_NAME: bosh-alicloud-kvm-ubuntu-xenial-go_agent
+          ensure:
+            do:
+              - <<: *teardown
+        ensure:
+          do:
+            - <<: *put-environment
+              params:
+                <<: *put-environment-params
+                env_name: certify-alicloud-bats-ubuntu-xenial
+                action: destroy
+                terraform_source: pipelines/alicloud/assets/terraform
+
+  - name: certify-ubuntu-xenial
+    serial: true
+    plan:
+      - aggregate:
+        - {get: bosh-release, trigger: true, passed: [bats-ubuntu-xenial]}
+        - {get: cpi-release,  trigger: true, passed: [bats-ubuntu-xenial]}
+        - {get: stemcell,     trigger: true, passed: [bats-ubuntu-xenial, test-stemcell-ubuntu-xenial], resource: ubuntu-xenial-stemcell}
+        - {get: pipelines,    trigger: false}
+      - task: generate
+        file: pipelines/shared/tasks/generate-receipt.yml
+        params:
+          CPI_RELEASE_NAME: bosh-alicloud-cpi
+          STEMCELL_NAME:    bosh-alicloud-kvm-ubuntu-xenial-go_agent
+        on_success:
+          put: notify
+          params:
+            text: "[bosh-cpi-certification Success]\nRunning task certify-ubuntu-xenial success."
+        on_failure:
+          put: notify
+          params:
+            text: "[bosh-cpi-certification Failed]\nRunning task certify-ubuntu-xenial failed. Please check it."
+      - {put: receipt, params: {file: certification/*-receipt.json}}
+
+resource_types:
+  - name: terraform_type
+    type: docker-image
+    source:
+      repository: ljfranklin/terraform-resource
+
+  - name: slack-notification
+    type: docker-image
+    source:
+      repository: cfcommunity/slack-notification-resource
+
+resources:
+  - name: after-midnight
+    type: time
+    source:
+      start: 00:00 AM
+      stop: 11:00 PM
+      location: {{time_location}}
+#  - name: bosh-cpi-src-in
+#    type: git
+#    source:
+#      uri: https://github.com/xiaozhu36/bosh-alicloud-cpi-release.git
+#      branch: master
+#      paths:
+#        - ci/**
+  - name: pipelines
+    type: git
+    source:
+      uri: https://github.com/aliyun/bosh-cpi-certification
+      branch: xiaozhu
+  - name: bosh-deployment
+    type: git
+    source:
+      uri: https://github.com/aliyun/bosh-deployment
+      branch: dev
+#  - name: cpi-release
```

Comment:
Can you please remove commented-out code?  We prefer to not have commented-out code in our configuration since it is confusing to people who read it.